### PR TITLE
SPLICE-2180 Fix code that uses Spark JobProgressListener.

### PR DESCRIPTION
**Pull request by stpabhi**

### CODE REVIEW

**Comment by jleach4** [comment]
@stpabhi Do you have a test case or at least a test scenario?  I would want someone to add something so we do not regress with new versions of spark for example.

**Comment by stpabhi** [comment]
@jleach4 This change is backward compatible with Spark 2.2. I introduced this change so that splice doesn't break when the runtime is Spark 2.3. I can work on a test case.

File: hbase_sql/src/main/java/com/splicemachine/olap/CompactionJob.java [hunk 1]
```diff
@@ -165,15 +166,15 @@ private boolean compactionRunning(List<Integer> jobIds) {
             return false;
```
Comment (dgomezferro) [inline hbase_sql/src/main/java/com/splicemachine/olap/CompactionJob.java]: This function checks whether the compaction actually started (or completed) execution, with this change it returns true whenever it is scheduled, but before its tasks start running. 

To use the new API we'd need to call getStageInfo() which has the same activeTasks and completedTasks information we were checking with the old API. You can get the stageId from the jobInfo object (there's only one stage for compactions) and then pass it to statusTracker.getStageInfo()

Comment (stpabhi) [reply hbase_sql/src/main/java/com/splicemachine/olap/CompactionJob.java]: Ok I misunderstood. Thanks for the clarification. Will make necessary changes.

**Comment by stpabhi** [comment]
@dgomezferro Made changes. please review.

**Comment by stpabhi** [comment]
@jyuanca Ok, will do.

**Comment by stpabhi** [comment]
@jyuanca #1997 #1998. Please review and approve.

**Pull request merged**